src/Adapter/PDF/OrderReturnPdfGenerator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\PDF;
+
+use Context;
+use OrderReturn;
+use PDF;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\OrderReturnSettings;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\PDF\PDFGeneratorInterface;
+use RuntimeException;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Validate;
+
+/**
+ * Generates invoice for given order.
+ */
+final class OrderReturnPdfGenerator implements PDFGeneratorInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generatePDF(array $orderReturnId)
```

Comment:
Why rely on this interface ? This interface look very wrong because it takes only one argument and is expecting an array... 

I think we should create a better interface here.
WDYT ?

Reply:
The whole src/Adapter/PDF needs to be refactored as a whole, but as you said this PR is already big enough so it's not a good idea to do it here. That's why for this I am following already established conventions.

Reply:
In this case, you can introduce a new interface in a dedicated PR, and rebase your work onto this one. We must not add any temporary code that we'll need to refactor later. It does not help with the migration.

Then you can rebase your work onto this one.

Reply:
IMO the PDF rework shouldn't block this PR, since the amount of code used here is rather small

Reply:
It is blocking, because the interface is wrong, and we should not use it anymore.
Because the signature contains an array, and we don't use an array, but an object. 

If you don't want to add a new interface here, I can be ok with this, but  don't use the old one.

And IMO, if it's small, then it's easy to add.

Reply:
Actually, it's not only the interface but the way PDF are generated (I believe it directly outputs on the stream) which is incorrect but we don't have a component to replace this yet So we should create a new component to generate PDF properly and replace the current usages. So using the old incorrect interface here is actually interesting for the future refacto as it will allow us to find all the use cases more easily